app/components/tailwinds/form/builder.rb

```diff
@@ -3,21 +3,51 @@
 module Tailwinds
   module Form
     # Provides Tailwind-styled forms
+    # :reek:InstanceVariableAssumption
     class Builder < Tramway::Views::FormBuilder
       def text_field(attribute, **options, &)
-        input = super(attribute, **options.merge(class: text_input_class))
-        render(Tailwinds::Form::TextFieldComponent.new(input, attribute, object_name:, **options), &)
+        render(Tailwinds::Form::TextFieldComponent.new(
+                 template: @template,
+                 attribute:,
+                 object_name:,
+                 label: label(options, attribute),
+                 for: for_id(attribute),
+                 options:
+               ), &)
       end
 
       def password_field(attribute, **options, &)
-        input = super(attribute, **options.merge(class: text_input_class))
-        render(Tailwinds::Form::TextFieldComponent.new(input, attribute, object_name:, **options), &)
+        render(Tailwinds::Form::TextFieldComponent.new(
+                 template: @template,
+                 attribute:,
+                 object_name:,
+                 label: label(options, attribute),
+                 for: for_id(attribute),
+                 options:
+               ), &)
       end
 
       def file_field(attribute, **options, &)
-        input = super(attribute, **options.merge(class: :hidden))
+        render(Tailwinds::Form::FileFieldComponent.new(
+                 template: @template,
+                 attribute:,
+                 object_name:,
+                 label: label(options, attribute),
+                 for: for_id(attribute),
+                 options:
+               ), &)
```

Comment:
seems like you could create method like

```
def default_options(attribute, options)
       {
                 template: @template,
                 attribute:,
                 object_name:,
                 label: label(options, attribute),
                 for: for_id(attribute),
                 options:
             }
end
```

and use it like

```
 render(Tailwinds::Form::FileFieldComponent.new(**default_options), &)
```

(maybe `render(Tailwinds::Form::FileFieldComponent.new(*default_options), &)`,
don't remember how many * are needed)

Reply:
done :slightly_smiling_face: